internal/controllers/clusterextension_controller.go

```diff
@@ -470,16 +472,31 @@ func (d *DefaultInstalledBundleGetter) GetInstalledBundle(ctx context.Context, e
 		return nil, err
 	}
 
-	release, err := cl.Get(ext.GetName())
+	rel, err := cl.Get(ext.GetName())
```

Comment:
Instead of continuing to use the `Get()` functionality here (I know this is all that helm-operator-plugins provides us), maybe we should:
- Update the [helm-operator-plugins `ActionInterface`](https://github.com/operator-framework/helm-operator-plugins/blob/1017ef84e14e4a35a69f946894d6387ae8a0180a/pkg/client/actionclient.go#L53) to have a `List()` method that creates a new [`List` type](https://pkg.go.dev/helm.sh/helm/v3@v3.16.1/pkg/action#List) and runs it with the provided options. The `List` type looks like it has the ability to be configured to filter releases by their various states and even sort them.
- Pull in the change from helm-operator-plugins and use the `List()` method here to:
    - Filter by "Deployed" releases
    - Sort by date (might need to reverse sort as well, I'm not sure if sorting by date would put newest first)

WDYT?

Reply:
Ah, the [`Sorter` type](https://pkg.go.dev/helm.sh/helm/v3@v3.16.1/pkg/action#Sorter) is a uint with a possible value of sorting by date descending

Reply:
You can filter by release name as well. Another thing probably worth verifying here, do we store history of the helm releases in our custom storage driver? If we don't, then should we?

Reply:
Also, should we be using only the ClusterExtension name or should we have something more deterministic based on ClusterExtension name + version?

Reply:
I think you might be getting a bit out-of-scope with history.